### PR TITLE
allow optional HTTP POST method override for logIn user method

### PR DESCRIPTION
**Pull request by kolohelios**

### CODE REVIEW

File: src/ParseUser.js [hunk 1]
```diff
@@ -439,7 +439,7 @@ class ParseUser extends ParseObject {
    * @return {Promise} A promise that is fulfilled with the user when
    *     the login is complete.
    */
-  logIn(options?: FullOptions): Promise<ParseUser> {
+  logIn(options?: FullOptions, usePost?: boolean): Promise<ParseUser> {
```
Comment (dplewis) [inline src/ParseUser.js:442]: Can you move `usePost` to options? Also add it to documentation.

File: src/__tests__/ParseUser-test.js [hunk 2]
```diff
@@ -229,6 +229,39 @@ describe('ParseUser', () => {
     });
   });
 
+  it('can log in as a user with POST method', (done) => {
```
Comment (dplewis) [inline src/__tests__/ParseUser-test.js:232]: Can you add an integration test?

**Review by dplewis — CHANGES_REQUESTED** [review]
Thanks for the PR! I was just thinking about this. I don't this will work without server update.

I added a fix last week [here](https://github.com/parse-community/parse-server/pull/6903). If you write an integration test it might fail.

File: src/ParseUser.js (hunk 1 above)
Comment (kolohelios) [reply src/ParseUser.js]: Done. Thank you so much for the quick review!

File: src/__tests__/ParseUser-test.js (hunk 2 above)
Comment (kolohelios) [reply src/__tests__/ParseUser-test.js:232]: I did, and it passed. I believe that the server is reading the credentials from the body as the short-circuit for query params is falsy though I could be wrong about that.

**Review by dplewis — APPROVED** [review]
LGTM!

**Pull request merged**